functions/HttpTriggerAPIUsersId/__init__.py

```diff
@@ -0,0 +1,53 @@
+import logging
+import pypyodbc
+import os
+import azure.functions as func
+import json
+
+
+def main(req: func.HttpRequest) -> func.HttpResponse:
+    db_server = os.environ["ENV_DATABASE_SERVER"]
+    db_name = os.environ["ENV_DATABASE_NAME"]
+    db_username = os.environ["ENV_DATABASE_USERNAME"]
+    db_password = os.environ["ENV_DATABASE_PASSWORD"]
+    driver = '{ODBC Driver 17 for SQL Server}'
+    connectionString = "Driver={};Server={};Database={};Uid={};Pwd={};Encrypt=yes;TrustServerCertificate=no;Connection Timeout=30;".format(
+        driver, db_server, db_name, db_username, db_password)
+
+    id = req.params.get('id')
+    if not id:
+        return func.HttpResponse(
+            "Bad Request",
+            status_code=400
+        )
+    else:
+        try:
+            with pypyodbc.connect(connectionString) as conn:
+                return execute_query(conn, id)
+        except pypyodbc.DatabaseError as e:
+            if e.args[0] == '28000':
+                return func.HttpResponse(
+                    "Unauthorized",
+                    status_code=403
+                )
+
+
+def execute_query(conn, id):
+    with conn.cursor() as cursor:
+        cursor.execute(
+            "SELECT * FROM users WHERE userId={}".format(id))
+        row = cursor.fetchone()
+        if not row:
+            return func.HttpResponse(
+                "User not found",
+                status_code=404
+            )
+        else:
+            columns = [column[0] for column in cursor.description]
+            data = dict(zip(columns, row))

```

Comment:
A comment here would be especially helpful. I'm guessing that "columns" is something like an array of all the field names and then row is the information? I'm just a bit confused as to exactly what's going on.

Reply:
This is something that I found on StackOverflow. This was tricky to get the column names shown in the json data. I will add a comment and link to the article